pkg/releasetesting/test_suite.go

```diff
@@ -69,53 +69,85 @@ func (ts *TestSuite) Run(env *Environment) error {
 		env.streamMessage("No Tests Found", release.TestRun_UNKNOWN)
 	}
 
+	var tests []*test
+
 	for _, testManifest := range ts.TestManifests {
 		test, err := newTest(testManifest)
 		if err != nil {
 			return err
 		}
 
-		test.result.StartedAt = timeconv.Now()
-		if err := env.streamRunning(test.result.Name); err != nil {
-			return err
+		tests = append(tests, test)
+	}
+
+	if env.Parallel {
+		c := make(chan error)
+		for _, t := range tests {
+			go func(t *test) {
+				c <- t.run(env)
+			}(t)
 		}
-		test.result.Status = release.TestRun_RUNNING
 
-		resourceCreated := true
-		if err := env.createTestPod(test); err != nil {
-			resourceCreated = false
-			if streamErr := env.streamError(test.result.Info); streamErr != nil {
+		for range tests {
```

Comment:
One thing I was thinking about - do we need/want to place a limit on the number of tests that can execute in parallel at one time?

My (slight) concern would be overloading the kubernetes api (or even the network), but maybe there's very low chance of that happening in practice.

Reply:
I don't think there should be limit built into helm - everyone's kubernetes setup is going to be different and will be able to handle different amounts of load. I could see maybe adding another flag for limiting the number of tests in parallel but I think that's scope creep at this point - would be better in a separate PR

Reply:
I agree with @mattjmcnaughton here. If we're parallelizing helm tests there should probably be a hard limit to the amount able to run at a time. I think that's well within the scope of adding parallelized support.